java/test/jmri/swing/PreferencesPanelTestBase.java

```diff
@@ -0,0 +1,99 @@
+package jmri.swing;
+
+import jmri.util.JUnitUtil;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.assertj.core.api.Java6Assertions.catchThrowable;
+
+/**
+ *  Base Tests for implementations of the PreferenesPanel interface.
+ *
+ * @author Paul Bender Colyright (C) 2020
+ */
+abstract public class PreferencesPanelTestBase {
+
+    protected PreferencesPanel prefsPanel;
+
+    /**
+     * Implementing classes should set the value of prefsPanel in setUp.
+     */
+    @BeforeEach
+    abstract public void setUp();
+
+    @AfterEach
+    public void tearDown(){
+        prefsPanel = null;
+        JUnitUtil.tearDown();
+    }
+
+    @Test
+    public void testCtor(){
+        assertThat(prefsPanel).isNotNull();
+    }
+
+    @Test
+    public void getPreferencesItem() {
+        assertThat(prefsPanel.getPreferencesItem()).isNotNull();
+    }
+
+    @Test
+    public void getPreferencesItemText() {
+        assertThat(prefsPanel.getPreferencesItemText()).isNotNull();
+    }
+
+    @Test
+    public void getTabbedPreferencesTitle() {
+        assertThat(prefsPanel.getTabbedPreferencesTitle()).isNotNull();
+    }
+
+    @Test
+    public void getLabelKey() {
+        assertThat(prefsPanel.getLabelKey()).isNotNull();
```

Comment:
Since the PreferencesPanel interface stipulates no requirement of nullness, this test should not assert is not null. If it is to assert anything, it should assert there is no exception.

Reply:
thanks, will fix